scylla/src/transport/connection.rs

```diff
@@ -1296,7 +1296,7 @@ impl Connection {
         &self,
         init_batch: &'b Batch,
         values: impl BatchValues,
-    ) -> Result<Cow<'b, Batch>, QueryError> {
+    ) -> Result<Cow<'b, Batch>, UserRequestError> {
         let mut to_prepare = HashSet::<&str>::new();
 
         {
```

Comment:
Commit: "batch: narrow the return error type of prepare_batch "

🤔 It is a bit surprising that this change does not require any other changes. This is due to some conversions done by "?", right? I wonder if we should maybe limit such conversion.
If error A has a variant with type B, then conversion from B to A is imo fine. But this is not the case here, right? This is a conversion between two separate errors. Maybe it should be explicit?

Reply:
`UserRequestError` has 11 variants. I think that `From<UserRequestError` for QueryError` impl comes in handy. I'm not a big fan of matching against all variants each time (assuming this is what you mean by explicit conversion).

Reply:
That is not what I meant. I'm talking specifically about `From` trait, because this is what seems to be used by `?` operator and `into` calls. 
A method like `into_query_error` would be totally fine.

Reply:
Okay, this makes sense. And I totally agree. Using `?` may lead to some unwanted implicit conversions.